src/main/java/org/kuali/kfs/gl/batch/service/impl/CuScrubberProcessImpl.java

```diff
@@ -0,0 +1,312 @@
+
+                    LOG.debug("generateCostShareEntries() Error 3 object not found");
+                    return new TransactionError(costShareSourceAccountEntry, m);
+                }
+
+                costShareSourceAccountOffsetEntry.setFinancialObjectCode(offsetDefinition.getFinancialObjectCode());
+                costShareSourceAccountOffsetEntry.setFinancialObject(offsetDefinition.getFinancialObject());
+                costShareSourceAccountOffsetEntry.setFinancialSubObjectCode(KFSConstants.getDashFinancialSubObjectCode());
+            }
+            else {
+                Map<Transaction, List<Message>> errors = new HashMap<Transaction, List<Message>>();
+
+                StringBuffer offsetKey = new StringBuffer("cost share transfer source ");
+                offsetKey.append(scrubbedEntry.getUniversityFiscalYear());
+                offsetKey.append("-");
+                offsetKey.append(scrubbedEntry.getChartOfAccountsCode());
+                offsetKey.append("-TF-");
+                offsetKey.append(scrubbedEntry.getFinancialBalanceTypeCode());
+
+                Message m = new Message(configurationService.getPropertyValueAsString(KFSKeyConstants.ERROR_OFFSET_DEFINITION_NOT_FOUND) + " (" + offsetKey.toString() + ")", Message.TYPE_FATAL);
+
+                LOG.debug("generateCostShareEntries() Error 4 offset not found");
+                return new TransactionError(costShareSourceAccountEntry, m);
+            }
+
+            costShareSourceAccountOffsetEntry.setFinancialObjectTypeCode(offsetDefinition.getFinancialObject().getFinancialObjectTypeCode());
+
+            if (scrubbedEntry.isCredit()) {
+                costShareSourceAccountOffsetEntry.setTransactionDebitCreditCode(KFSConstants.GL_DEBIT_CODE);
+            }
+            else {
+                costShareSourceAccountOffsetEntry.setTransactionDebitCreditCode(KFSConstants.GL_CREDIT_CODE);
+            }
+
+            try {
+                flexibleOffsetAccountService.updateOffset(costShareSourceAccountOffsetEntry);
+            }
+            catch (InvalidFlexibleOffsetException e) {
+                Message m = new Message(e.getMessage(), Message.TYPE_FATAL);
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("generateCostShareEntries() Cost Share Transfer Account Flexible Offset Error: " + e.getMessage());
+                }
+                return new TransactionError(costShareEntry, m);
+            }
+
+            createOutputEntry(costShareSourceAccountOffsetEntry, OUTPUT_GLE_FILE_ps);
+            scrubberReport.incrementCostShareEntryGenerated();
+
+            scrubCostShareAmount = KualiDecimal.ZERO;
+        } catch (IOException ioe) {
+            LOG.error("generateCostShareEntries() Stopped: " + ioe.getMessage());
+            throw new RuntimeException("generateCostShareEntries() Stopped: " + ioe.getMessage(), ioe);
+        }

```

Comment:
If this exception is thrown, will there be enough output to the log(s) to troubleshoot and fix the root cause of the exception?